### PR TITLE
[#5548] feat(core): add fileset pre event to Gravitino event 

### PR DESCRIPTION
**Pull request by TungYuChiang**
### What changes were proposed in this pull request?

add fileset pre event to Gravitino event 

### Why are the changes needed?

Fix: #5548 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

added unit test

### CODE REVIEW

File: core/src/main/java/org/apache/gravitino/listener/api/event/AlterFilesetPreEvent.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * under the License.
+ */
+
+package org.apache.gravitino.listener.api.event;
+
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.annotation.DeveloperApi;
+import org.apache.gravitino.file.FilesetChange;
+
+@DeveloperApi
```
Comment (FANNG1) [inline core/src/main/java/org/apache/gravitino/listener/api/event/AlterFilesetPreEvent.java:26]: Could you add java doc for the class?

**Comment by FANNG1** [comment]
LGTM except one minor comment, @TungYuChiang could you fix it?

**Comment by TungYuChiang** [comment]
Thank you for your review!  I've add the document

File: core/src/main/java/org/apache/gravitino/listener/api/event/AlterFilesetPreEvent.java (hunk 1 above)
Comment (FANNG1) [reply core/src/main/java/org/apache/gravitino/listener/api/event/AlterFilesetPreEvent.java:26]: Could you add `/** Represents an event triggered before altering a fileset. */`  before the class ?

Comment (TungYuChiang) [reply core/src/main/java/org/apache/gravitino/listener/api/event/AlterFilesetPreEvent.java:27]: Apologies for the earlier misunderstanding. I have now added the suggested JavaDoc before the class,

File: core/src/main/java/org/apache/gravitino/listener/FilesetEventDispatcher.java [hunk 2]
```diff
@@ -32,16 +32,22 @@
...
+import org.apache.gravitino.listener.api.event.DropFilesetPreEvent;
 import org.apache.gravitino.listener.api.event.GetFileLocationEvent;
 import org.apache.gravitino.listener.api.event.GetFileLocationFailureEvent;
+import org.apache.gravitino.listener.api.event.GetFileLocationPreEvent;
 import org.apache.gravitino.listener.api.event.ListFilesetEvent;
 import org.apache.gravitino.listener.api.event.ListFilesetFailureEvent;
+import org.apache.gravitino.listener.api.event.ListFilesetPreEvent;
 import org.apache.gravitino.listener.api.event.LoadFilesetEvent;
 import org.apache.gravitino.listener.api.event.LoadFilesetFailureEvent;
+import org.apache.gravitino.listener.api.event.LoadFilesetPreEvent;
```
Comment (tengqm) [inline core/src/main/java/org/apache/gravitino/listener/FilesetEventDispatcher.java:50]: Is `import org.apache.gravitino.listener.api.event.*` anti-pattern thus discouraged?

File: core/src/main/java/org/apache/gravitino/listener/FilesetEventDispatcher.java [hunk 3]
```diff
@@ -96,6 +104,10 @@ public Fileset createFileset(
       String storageLocation,
       Map<String, String> properties)
       throws NoSuchSchemaException, FilesetAlreadyExistsException {
+    FilesetInfo createFileRequest =
+        new FilesetInfo(ident.name(), comment, type, storageLocation, properties, null);
+    eventBus.dispatchEvent(
+        new CreateFilesetPreEvent(PrincipalUtils.getCurrentUserName(), ident, createFileRequest));
```
Comment (tengqm) [inline core/src/main/java/org/apache/gravitino/listener/FilesetEventDispatcher.java:110]:
```suggestion
    FilesetInfo fileInfo =
        new FilesetInfo(ident.name(), comment, type, storageLocation, properties, null);
    eventBus.dispatchEvent(
        new CreateFilesetPreEvent(PrincipalUtils.getCurrentUserName(), ident, fileInfo));
```

File: core/src/main/java/org/apache/gravitino/listener/api/event/CreateFilesetPreEvent.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  /**
+   * Retrieves the create fileset request.
+   *
+   * @return A {@link FilesetInfo} instance encapsulating the comprehensive details of create
+   *     fileset request.
+   */
+  public FilesetInfo createFilesetRequest() {
+    return createFilesetRequest;
```
Comment (tengqm) [inline core/src/main/java/org/apache/gravitino/listener/api/event/CreateFilesetPreEvent.java:44]: What you are returning here is not the full set of parameters for a CreateFileset request, so the info you retrieved should be a filesetInfo as the return type of the function indicates.

```suggestion
    return fileInfo;
```

File: core/src/main/java/org/apache/gravitino/listener/api/event/AlterFilesetPreEvent.java (hunk 1 above)
Comment (FANNG1) [reply core/src/main/java/org/apache/gravitino/listener/api/event/AlterFilesetPreEvent.java:27]: Never mind, thanks for your contribution

File: core/src/main/java/org/apache/gravitino/listener/FilesetEventDispatcher.java (hunk 2 above)
Comment (FANNG1) [reply core/src/main/java/org/apache/gravitino/listener/FilesetEventDispatcher.java:50]: yes, import xx.* is not preferred

**Review by FANNG1 — APPROVED** [review]

File: core/src/main/java/org/apache/gravitino/listener/api/event/CreateFilesetPreEvent.java (hunk 4 above)
Comment (FANNG1) [reply core/src/main/java/org/apache/gravitino/listener/api/event/CreateFilesetPreEvent.java:44]: I'd like to keep the original naming, for it's more meaningful to me.

**Comment by FANNG1** [comment]
merged to main, thanks @TungYuChiang  for your work, and @tengqm for reviewing!

**Pull request merged**